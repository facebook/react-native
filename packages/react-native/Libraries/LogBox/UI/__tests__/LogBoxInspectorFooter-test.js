/**
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 * @oncall react_native
 */

'use strict';

const render = require('../../../../jest/renderer');
const LogBoxInspectorFooter = require('../LogBoxInspectorFooter').default;
const React = require('react');

// Mock `LogBoxInspectorFooterButton` because we are interested in snapshotting
// the behavior of `LogBoxInspectorFooter`, not `LogBoxInspectorFooterButton`.
jest.mock('../LogBoxInspectorFooterButton', () => ({
  __esModule: true,
  default: 'LogBoxInspectorFooterButton',
}));

describe('LogBoxInspectorFooter', () => {
  it('should render two buttons for warning', async () => {
    const output = await render.create(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="warn"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render two buttons for error', async () => {
    const output = await render.create(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="error"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render two buttons for fatal', async () => {
    const output = await render.create(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="fatal"
      />,
    );

    expect(output).toMatchSnapshot();
  });

  it('should render no buttons and a message for syntax error', async () => {
    const output = await render.create(
      <LogBoxInspectorFooter
        onMinimize={() => {}}
        onDismiss={() => {}}
        level="syntax"
      />,
    );

    expect(output).toMatchSnapshot();
  });
});
