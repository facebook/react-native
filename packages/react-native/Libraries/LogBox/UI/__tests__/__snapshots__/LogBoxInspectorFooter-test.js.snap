// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogBoxInspectorFooter should render no buttons and a message for syntax error 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "flexDirection": "row",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": -2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgba(255, 255, 255, 0.6)",
          "fontSize": 14,
          "fontStyle": "italic",
          "height": 48,
          "lineHeight": 20,
          "paddingBottom": 50,
          "paddingTop": 20,
          "textAlign": "center",
          "width": "100%",
        }
      }
    >
      This error cannot be dismissed.
    </Text>
  </View>
</View>
`;

exports[`LogBoxInspectorFooter should render two buttons for error 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "flexDirection": "row",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": -2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <LogBoxInspectorFooterButton
    onPress={[Function]}
    text="Dismiss"
  />
  <LogBoxInspectorFooterButton
    onPress={[Function]}
    text="Minimize"
  />
</View>
`;

exports[`LogBoxInspectorFooter should render two buttons for fatal 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "flexDirection": "row",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": -2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <LogBoxInspectorFooterButton
    onPress={[Function]}
    text="Dismiss"
  />
  <LogBoxInspectorFooterButton
    onPress={[Function]}
    text="Minimize"
  />
</View>
`;

exports[`LogBoxInspectorFooter should render two buttons for warning 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(51, 51, 51, 1)",
      "flexDirection": "row",
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": -2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 2,
    }
  }
>
  <LogBoxInspectorFooterButton
    onPress={[Function]}
    text="Dismiss"
  />
  <LogBoxInspectorFooterButton
    onPress={[Function]}
    text="Minimize"
  />
</View>
`;
